src/map/map_info.rs

```diff
@@ -115,51 +127,92 @@ impl MapInfo {
                     .collect();
 
                 let mut group = Group::new().set("class", get_class_names(&css));
-                for entry in &entries {
+                for (index, entry) in entries.iter().enumerate() {
                     if let Some(path) =
                         points_to_svg_path(&entry.points, entry.close_path, force_connected)
                     {
+                        let path = if colorize {
+                            let color_class = ROOM_COLORS[index % 4];
+                            used_styles.insert(color_class);
+                            path.set("class", get_style(&color_class).class_name)
+                        } else {
+                            path
+                        };
                         group = group.add(path);
                     }
                 }
                 svg_elements.push(Box::new(group));
-                used_styles.insert(css);
+                used_styles.extend(css);
                 if map_info_type == MapInfoType::Outline {
                     viewbox = calc_viewbox(&entries);
                 }
             }
         }
 
-        Some((
-            svg_elements,
-            viewbox?,
-            used_styles.into_iter().flatten().collect(),
-        ))
+        // Generate additional required defs elements
+        let mut added_defs: Vec<Box<dyn svg::node::Node>> = Vec::new();
+        if used_styles.contains(&CSSClass::RoomUnreachable) {
+            // Diagonal stripes pattern (for unreachable areas)
+            let pattern = Pattern::new()
+                .set("id", "ds")
+                .set("x", 0)
+                .set("y", 0)
+                .set("width", 2)
+                .set("height", 2)
+                .set("patternUnits", "userSpaceOnUse")
+                .set("patternTransform", "rotate(45)")
+                .add(
+                    Line::new()
+                        .set("x1", 0)
+                        .set("y1", 0)
+                        .set("x2", 0)
+                        .set("y2", 2)
+                        .set("stroke", "rgba(0, 0, 0, 0.2)")
+                        .set("stroke-width", 1),
+                );
+            added_defs.push(Box::new(pattern));
+        }
+
+        Some((svg_elements, viewbox?, used_styles, added_defs))
     }
 
-    fn get_order(&self) -> [(MapInfoType, Vec<CSSClass>, bool); 3] {
+    fn get_order(&self) -> Vec<(MapInfoType, Vec<CSSClass>, bool, bool)> {
```

Comment:
We should create a struct (instead of a tuple) for better readability

Reply:
Done